### PR TITLE
4032 Tweaked V4 RECAP Search API objects

### PR DESCRIPTION
**Pull request by albertisfu**
According to #4032 I've applied the following changes in this PR:

- Removed `court_exact` from the response object in all search RECAP search types.
- Added a new `meta` key into all the RECAP search types that contain:
  - r type:
    -  docket level `meta`
       -  more_docs
       - timestamp (timestamp when the docket was indexed)
       - date_created (docket date_created)
     - recap_document level `meta`
        - timestamp (timestamp when rd was indexed)
       - date_created (rd date_created)
 - d type `meta`:
    - timestamp (timestamp when the docket was indexed)
    - date_created (docket date_created)
 
 - rd type `meta`:
    - timestamp (timestamp when rd was indexed)
    - date_created (rd date_created)

Here some examples:

### r type:

```
{
   "assignedTo":null,
   "assigned_to_id":null,
   "attorney":[
      
   ],
   "attorney_id":[
      
   ],
   "caseName":"Attia v. Oura Ring, Inc.",
   "case_name_full":"",
   "cause":"28:1332 Diversity-Breach of Contract",
   "chapter":null,
   "court":"District Court, N.D. California",
   "court_citation_string":"N.D. Cal.",
   "court_id":"cand",
   "dateArgued":null,
   "dateFiled":null,
   "dateTerminated":null,
   "docketNumber":"4:23-cv-0343324",
   "docket_absolute_url":"/docket/193/attia-v-oura-ring-inc/",
   "docket_id":193,
   "firm":[
      
   ],
   "firm_id":[
      
   ],
   "jurisdictionType":"",
   "juryDemand":"",
   "meta":{
      "timestamp":"2024-05-07T17:07:32.966814Z",
      "date_created":"2024-04-12T17:17:49.884083Z",
      "more_docs":false
   },
   "pacer_case_id":"415205",
   "party":[
      
   ],
   "party_id":[
      
   ],
   "recap_documents":[
      {
         "absolute_url":"",
         "attachment_number":null,
         "cites":[
            
         ],
         "description":"ORDER DENYING DEFENDANTS'   16  MOTION TO COMPEL ARBITRATION AND TO DISMISS OR, IN THE ALTERNATIVE, STAY PROCEEDINGS.Case Management Statement due by 4/9/2024. ",
         "docket_entry_id":15083,
         "document_number":null,
         "document_type":"PACER Document",
         "entry_date_filed":"2024-04-01",
         "entry_number":null,
         "filepath_local":null,
         "id":15084,
         "is_available":false,
         "meta":{
            "timestamp":"2024-05-07T17:07:33.011149Z",
            "date_created":"2024-04-12T17:17:49.910354Z"
         },
         "pacer_doc_id":"",
         "page_count":null,
         "short_description":"",
         "snippet":""
      }
   ],
   "referredTo":null,
   "referred_to_id":null,
   "suitNature":"Contract: Other",
   "trustee_str":null
}
```

### rd type:

```
{
   "absolute_url":"",
   "attachment_number":null,
   "cites":[
      
   ],
   "description":"ORDER DENYING DEFENDANTS'   16  MOTION TO COMPEL ARBITRATION AND TO DISMISS OR, IN THE ALTERNATIVE, STAY PROCEEDINGS.Case Management Statement due by 4/9/2024.,
   "docket_entry_id":15083,
   "docket_id":193,
   "document_number":null,
   "document_type":"PACER Document",
   "entry_date_filed":"2024-04-01",
   "entry_number":null,
   "filepath_local":null,
   "id":15084,
   "is_available":false,
   "meta":{
      "timestamp":"2024-05-07T17:07:33.011149Z",
   
   "date_created":"2024-04-12T17:17:49.910354Z"
   },
   "pacer_doc_id":"",
   "page_count":null,
   "short_description":"",
   "snippet":""
}
```

### d type:

```
{
   "assignedTo":null,
   "assigned_to_id":null,
   "attorney":[
      
   ],
   "attorney_id":[
      
   ],
   "caseName":"Attia v. Oura Ring, Inc.",
   "case_name_full":"",
   "cause":"28:1332 Diversity-Breach of Contract",
   "chapter":null,
   "court":"District Court, N.D. California",
   "court_citation_string":"N.D. Cal.",
   "court_id":"cand",
   "dateArgued":null,
   "dateFiled":null,
   "dateTerminated":null,
   "docketNumber":"4:23-cv-0343324",
   "docket_absolute_url":"/docket/193/attia-v-oura-ring-inc/",
   "docket_id":193,
   "firm":[
      
   ],
   "firm_id":[
      
   ],
   "jurisdictionType":"",
   "juryDemand":"",
   "meta":{
      "timestamp":"2024-05-07T17:07:32.966814Z",
      "date_created":"2024-04-12T17:17:49.884083Z"
   },
   "pacer_case_id":"415205",
   "party":[
      
   ],
   "party_id":[
      
   ],
   "referredTo":null,
   "referred_to_id":null,
   "suitNature":"Contract: Other",
   "trustee_str":null
}
```

Let me know if additional changes to the objects are needed.

### CODE REVIEW

**Review by mlissner — COMMENTED** [review]
Looks generally good to me, though that's a whole lot of inheritance. I'd feel better if we did it as mixins, and I think that'd work fine too? Just a thought.

**Comment by albertisfu** [comment]
Great, yeah, mixins might be better here. I'll give it a try.

**Comment by albertisfu** [comment]
I've applied the suggestion and added a couple of mixins where it was possible: `MetaMixin` and `RECAPTypeMetaMixin`, to simplify the common field inheritance.

File: cl/search/api_serializers.py [hunk 1]
```diff
@@ -432,11 +432,41 @@ class Meta:
         )
 
 
-class BaseRECAPDocumentESResultSerializer(DocumentSerializer):
+class NestedMetaSerializer(serializers.Serializer):
```
Comment (ERosendo) [inline cl/search/api_serializers.py:435]: I believe the word 'nested' in the name of this class might be redundant. We have other serializer classes to handle complex objects with varying levels of nesting, but their names don't explicitly mention it. Maybe we can rename it to  "MetaDataSerializer",  What do you think?

Comment (albertisfu) [reply cl/search/api_serializers.py]: Yeah, I agree, thanks.
I've applied the suggestion.

**Comment by albertisfu** [comment]
Thanks, Eduardo, for your suggestion. I've applied the classes name change. @mlissner, this is ready to merge unless you have additional comments

**Comment by mlissner** [comment]
Great! Thank you both! This is a nice little improvement!

**Pull request merged**